dev/camdenhells/REQUIREMENTS.md

```diff
@@ -105,15 +108,16 @@ Provide separate specification of a rendered view for a scene model. Each scene
 2. One camera, switchable, and a dynamic set of any number/mix of lights
   * No inherent geometry/vertex data
   * Embedded in the scene, transformable 
-  * World coordinates can be querie
+  * World coordinates can be queried
   * "Containers", optional visualization geometries for debugging or app functions can be added
   * Switch betwen whole light sets at once ? [2.1]
-3. Skybox texture/bitmaps and size, and/or background color; all mutable
+3. Skybox texture/bitmaps and size, and/or background color and/or materials; all mutable
```

Comment:
Why are materials added separately here?  This bullet should have been just about defining the background of the scene view, i.e. a skybox and/or a clear color, but I threw in the Skybox parameters as well. 
We should probably have a separate  requirement for any skybox enhancements per se,

Are you using "material" to mean "program"? If so, I want to advocate for consistently distinguishing throughout CH between material specs and programs (some of which implement material/lighting specs). Or do you mean that skyboxes should support material/lighting specs in addition to textures (i.e. "skybox texture/bitmaps/_material_spec and size, and/or background color")?  If so, are the skybox material spec and texture/bitmaps mutually exclusive, and is the material spec per cube face? 

BTW,  I'd like to cleanly separate a Skybox child from the Cube primitive. So if the goal is to allow material specs then we should probably modify the Cube instead to accept a material (face array?)...

Reply:
- `ObjectRenderer` is what I meant
- I'm fine with a Skybox/Skysphere primitive
- I need to do some research before I can make a decision on the per face textures.

Reply:
Yep to the ObjectRenderer for skyboxes...  Exactly what I had in mind too.  I'd probably move the skybox/sphere out of the primitives package and into the sceneview package, as they make more sense to me as an integral part of a SceneView rather than as general primitives, and they track the Camera location, etc.   Not worried about per face stuff for now.

Reply:
When messing with skybox/sphere we also need to keep Environment Mapping in mind. It is always embarrassing when reflective materials do not reflect the sky.

Reply:
Are you suggesting that when a Skybox is active any environment map materials should Automatically update?

Reply:
Currently, we have cube and sphere environment mapping. If we are going to add an equirectangular mapped skyshere geometry, we should also add equirectangular environment mapping so we can continue to share textures between both sky geometry and environment mapping.